### PR TITLE
upgrade to Scala 2.12.12 from 2.12.11

### PR DESCRIPTION
**Pull request by S11001001**
Upgrading doobie to the latest release, as well as related libraries, will push the scala-library dependency to 2.12.12, so doing that separately and consistently with the compiler version.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Thank you!

**Review by remyhaemmerle-da — APPROVED** [review]
Nice. Thanks

**Review by SamirTalwar-DA — APPROVED** [review]
Looks great to me!

**Review by aherrmann-da — APPROVED** [review]
LGTM, thank you!

File: daml-lf/interpreter/BUILD.bazel [hunk 1]
```diff
@@ -13,7 +13,7 @@ da_scala_library(
     name = "interpreter",
     srcs = glob(["src/main/**/*.scala"]),
     plugins = [
-        "@maven//:com_github_ghik_silencer_plugin_2_12_11",
+        "@maven//:com_github_ghik_silencer_plugin_2_12_12",
```
Comment (S11001001) [inline daml-lf/interpreter/BUILD.bazel:16]: For future reference, #7668 has captured all changes to these silencer strings in scala.bzl.

**Pull request merged**